presto-yarn-test/src/test/groovy/com/teradata/presto/yarn/PrestoClusterTest.groovy

```diff
@@ -124,6 +131,40 @@ class PrestoClusterTest
   }
 
   @Test
+  void 'single node presto app - adding plugin'()
+  {
+    PrestoCluster prestoCluster = new PrestoCluster(slider, hdfsClient, 'resources-singlenode.json', APP_CONFIG_WITHOUT_CATALOG_TEMPLATE)
+    prestoCluster.withPrestoCluster {
+      prestoCluster.assertThatPrestoIsUpAndRunning(0)
+
+      assertThatAllProcessesAreRunning(prestoCluster)
+
+      assertThatPluginJarExists(prestoCluster)
+
+      assertThatApplicationIsStoppable(prestoCluster)
+    }
+  }
+
+  def assertThatPluginJarExists(PrestoCluster prestoCluster) {
```

Comment:
Even simpler is a plugin that provides a function. Look at presto-ml for an example.